[DO NOT MERGE YET] Move to go1.8

vadmeste opened the pull request:
## Description
Support of go 1.8

Once all issues under POD move to go 1.8 are closed, we can merge this PR. 

## Motivation and Context
POD move to go 1.8

## How Has This Been Tested?
go test + Manully

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

harshavardhana committed:
Move CI builds to go1.8

vadmeste committed:
1.8: Fix failed tests

fwessels commented on cmd/posix-prepare-path.go:
```diff
@@ -16,41 +16,7 @@
-)
-
 // preparePath rewrites path to handle any OS specific details.
 func preparePath(path string) string {
```
Remove function altogether?

vadmeste replied:
> Remove function altogether?

Done

vadmeste committed:
1.8: Workround for deadline + hijack go1.8 bug

This is a workaround for this golang bug
https://github.com/golang/go/issues/21133

vadmeste committed:
1.8: Remove preparePath() function

As it is not needed anymore with go 1.8

harshavardhana commented on cmd/endpoint_test.go:
```diff
@@ -232,7 +232,7 @@ func TestCreateEndpoints(t *testing.T) {
 		expectedErr        error
 	}{
-		{"localhost", []string{}, "", EndpointList{}, -1, fmt.Errorf("missing port in address localhost")},
+		{"localhost", []string{}, "", EndpointList{}, -1, fmt.Errorf("address localhost: missing port in address")},
```
We should remove these style of error checks.. in future.

harshavardhana commented on pkg/http/quirkconn.go:
```diff
@@ -0,0 +1,45 @@
+}
+
+// SetReadDeadline - implements a workaround of SetReadDeadline go bug
+func (q *QuirkConn) SetReadDeadline(t time.Time) error {
```
We should add a test for this.. Looks like currently this is not used. Perhaps you can also rename the file as ```conn_bug_21133.go```  and makes it easy to delete.

vadmeste replied:
@harshavardhana yeah we are right, it was a mistake and I fixed it..

By the way, I really tried to make QuirkConn fully transparent.. but it wasn't possible, bufConn should be aware about the read deadline problem and should check by itself if it is allowed to set read deadline to avoid causing the go 1.8 bug.

The reason is that both golang API and bufConn calls SetReadDeadline() and we really need to synchronize those calls due to the go 1.8 bug.

vadmeste committed:
go1.8: Add canSetReadDeadline() to QuirkConn

There is no way to have a transparent QuirkConn which provides a
workaround of this golang/go#21133, the easiest way is to expose
canSetReadDeadline() API to test if we are allowed to call
SetReadDealine API without triggering the golang issue.

vadmeste committed:
Enhance minikube example

vadmeste committed:
go1.8: Fix tests failure for Windows
